backend/app/services/create_cap_table.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+class CreateCapTable
+  def initialize(company:, investors_data:)
+    @company = company
+    @investors_data = investors_data
+    @errors = []
+  end
+
+  def perform
+    return { success: false, errors: ["Company must have equity enabled"] } unless company.equity_enabled?
+    # Check if company already has cap table data
+    existing_cap_table_errors = validate_no_existing_cap_table
+    if existing_cap_table_errors.any?
+      return { success: false, errors: ["Company already has cap table data: #{existing_cap_table_errors.to_sentence}"] }
+    end
+
+    validate_data
+    return { success: false, errors: @errors } if @errors.any?
+
+    ApplicationRecord.transaction do
+      create_share_class_if_needed
+      create_investors_and_holdings
+    end
+
+    # Update company shares after transaction commits (so ShareHolding callbacks have run)
+    update_company_shares
+
+    { success: true, errors: [] }
+  rescue ActiveRecord::RecordInvalid => e
+    @errors << e.message
+    { success: false, errors: @errors }
+  rescue => e
+    @errors << "Unexpected error: #{e.message}"
+    { success: false, errors: @errors }
+  end
+
+  private
+    attr_reader :errors, :company, :investors_data
+
+    def validate_no_existing_cap_table
+      errors = []
+      errors << "option pools" if company.option_pools.exists?
+      errors << "share classes" if company.share_classes.exists?
+      errors << "investors" if company.company_investors.exists?
+      errors << "share holdings" if company.share_holdings.exists?
+      errors
+    end
+
+    def validate_data
+      total_shares = 0
+      investors_data.each_with_index do |investor_data, index|
+        user = User.find_by(external_id: investor_data[:userId])
+        shares = investor_data[:shares].to_i
+
+        if user.nil?
+          @errors << "Investor #{index + 1}: User not found"
+          next
+        end
+
+        if company.company_investors.exists?(user: user)
+          @errors << "Investor #{index + 1}: User is already an investor in this company"
+          next
+        end
+
+        total_shares += shares
+      end
+
+      if company.fully_diluted_shares > 0 && total_shares > company.fully_diluted_shares
+        @errors << "Total shares (#{total_shares}) cannot exceed company's fully diluted shares (#{company.fully_diluted_shares})"
+      end
+    end
+
+    def create_share_class_if_needed
+      return if company.share_classes.exists?(name: ShareClass::DEFAULT_NAME)
```

Comment:
```suggestion
    def create_share_class
```

As we're always going to create it now.